### PR TITLE
[JENKINS-49779] Feature/sequentiel add more tests

### PR DESCRIPTION
**Pull request by olamy**
# Description

See [JENKINS-49779](https://issues.jenkins-ci.org/browse/JENKINS-49779).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java [hunk 1]
```diff
@@ -260,13 +260,17 @@ protected void handleChunkDone(@Nonnull MemoryFlowChunk chunk) {
             }
         }else{
             if(parallelNestedStages) {
-                String endId = parallelBranchEndNodes.peek().getId();
-                Stack<FlowNodeWrapper> stack = stackPerEnd.get(endId);
-                if(stack==null){
-                    stack=new Stack<>();
-                    stackPerEnd.put(endId, stack);
+                if(parallelBranchEndNodes.isEmpty()){
+                    logger.warn("skip parsing stage {} but parallelBranchEndNodes is empty", stage);
```
Comment (vivek) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java]: Do we want this to be warning or debug level? In which condition we will hit this case? Making sure what this warning level means in terms of identifying underlying issue.

Comment (olamy) [reply blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java]: oops. I added as warning level to ease debug with @NicuPascu . Sure debug might be better.
Keep as it and I will change that once everything is ok for @NicuPascu.
So we can wait before merging this pr.

**Comment by olamy** [comment]
@vivek ready to merge now. @NicuPascu seems happy now :)

**Comment by NicuPascu** [comment]
Thanks for the fixes and the extra field :bee:

**Comment by vivek** [comment]
@olamy great, there is leftover commented code and a warning log message that  should  be debug level?

**Comment by olamy** [comment]
@vivek good catch I missed that. All done (I know let's wait the end of CI build :P )

File: blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineNode.java [hunk 2]
```diff
@@ -79,6 +80,12 @@
     @Exported
     public abstract boolean isRestartable();
 
+    /**
+     * @return the id of the first parent
+     */
+    @Exported
```
Comment (vivek) [inline blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineNode.java]: @olamy I missed this. ‘getFirstParent()’ is ambiguous. Is it first immediate parent or first node in the DAG?

**Pull request merged**